### ISSUE TITLE
Dockerhub build fails (but mybinder works and build image locally)

### ISSUE THREAD

**Issue by fzeiser** [oslocyclotronlab/ompy#96]
Dockerhub build started to fail some time ago (but mybinder works and build image locally). The build log is :

```
Cloning into '.'...
Warning: Permanently added the RSA host key for IP address '140.82.113.3' to the list of known hosts.
Submodule 'oscar_response' (https://github.com/oslocyclotronlab/oscar_response.git) registered for path 'oscar_response'
Cloning into 'oscar_response'...
Submodule path 'oscar_response': checked out 'dc520845341affdb6547dc17e778b60899f6b20c'
Reset branch 'master'
Your branch is up-to-date with 'origin/master'.
Pulling cache layers for index.docker.io/oslocyclotronlab/ompy:latest...
Done!
KernelVersion: 4.4.0-1060-aws
Components: [{u'Version': u'18.03.1-ee-3', u'Name': u'Engine', u'Details': {u'KernelVersion': u'4.4.0-1060-aws', u'Os': u'linux', u'BuildTime': u'2018-08-30T18:42:30.000000000+00:00', u'ApiVersion': u'1.37', u'MinAPIVersion': u'1.12', u'GitCommit': u'b9a5c95', u'Arch': u'amd64', u'Experimental': u'false', u'GoVersion': u'go1.10.2'}}]
Arch: amd64
BuildTime: 2018-08-30T18:42:30.000000000+00:00
ApiVersion: 1.37
Platform: {u'Name': u''}
Version: 18.03.1-ee-3
MinAPIVersion: 1.12
GitCommit: b9a5c95
Os: linux
GoVersion: go1.10.2
Starting build of index.docker.io/oslocyclotronlab/ompy:latest...
Step 1/12 : FROM jupyter/minimal-notebook:latest

[...]

Step 10/12 : USER $NB_USER
---> Running in a58ead4ead79
Removing intermediate container a58ead4ead79
---> 5d67b3535740
Step 11/12 : COPY --chown=1000:100 . ompy
---> ee0691472363
Step 12/12 : RUN cd ompy && git submodule update --init --recursive && pip install -e .
---> Running in 88a231fdc76c
fatal: not a git repository: /src/bykexuscmgfuhke6khq9e4h/.git/modules/oscar_response
Unable to find current revision in submodule path 'oscar_response'
Removing intermediate container 88a231fdc76c
The command '/bin/sh -c cd ompy && git submodule update --init --recursive && pip install -e .' returned a non-zero code: 1
```

so in essence the error is 
```
fatal: not a git repository: /src/bykexuscmgfuhke6khq9e4h/.git/modules/oscar_response
Unable to find current revision in submodule path 'oscar_response'
```

I was able to fix it with the solution from:
https://stackoverflow.com/questions/58690455/how-to-correctly-initialize-git-submodules-in-dockerfile-for-docker-cloud

where

> TL;DR: Place this in hooks/post_checkout:
> 

> ```
> #!/bin/bash
> # Docker hub does a recursive clone, then checks the branch out,
> # so when a PR adds a submodule (or updates it), it fails.
> git submodule update --init
> ```
> 

and removing the `git submodule update --init` from the Dockerfile.

Proposed changed to the `fix/dockerhub` branch and it seems to work.

**Comment by fzeiser** [comment]
Build log now instead:
```
Cloning into '.'...
Warning: Permanently added the RSA host key for IP address '192.30.253.113' to the list of known hosts.
Submodule 'oscar_response' (https://github.com/oslocyclotronlab/oscar_response.git) registered for path 'oscar_response'
Cloning into 'oscar_response'...
Submodule path 'oscar_response': checked out 'dc520845341affdb6547dc17e778b60899f6b20c'
Switched to a new branch 'fix/dockerhub'
Executing post_checkout hook...
KernelVersion: 4.4.0-1060-aws
Components: [{u'Version': u'18.03.1-ee-3', u'Name': u'Engine', u'Details': {u'KernelVersion': u'4.4.0-1060-aws', u'Os': u'linux', u'BuildTime': u'2018-08-30T18:42:30.000000000+00:00', u'ApiVersion': u'1.37', u'MinAPIVersion': u'1.12', u'GitCommit': u'b9a5c95', u'Arch': u'amd64', u'Experimental': u'false', u'GoVersion': u'go1.10.2'}}]
Arch: amd64
BuildTime: 2018-08-30T18:42:30.000000000+00:00
ApiVersion: 1.37
Platform: {u'Name': u''}
Version: 18.03.1-ee-3
MinAPIVersion: 1.12
GitCommit: b9a5c95
Os: linux
GoVersion: go1.10.2
Starting build of index.docker.io/oslocyclotronlab/ompy:latest...
Step 1/12 : FROM jupyter/minimal-notebook:latest

[...]

Step 10/12 : USER $NB_USER
---> Running in 4b6fdfe3ac2c
Removing intermediate container 4b6fdfe3ac2c
---> 45b3b03ca83e
Step 11/12 : COPY --chown=1000:100 . ompy
---> 23d04a6596eb
Step 12/12 : RUN cd ompy && pip install -e .
---> Running in fec0fb794526
[...]
Done!
Build finished
